In [1]:
import numpy as np
import pandas as pd

Q1. Pandas version

What's the version of Pandas that you installed?

In [5]:
pd.__version__

'2.1.4'

In [25]:
data = pd.read_csv('laptops.csv')

Q2. Records count

How many records are in the dataset?

In [29]:
data.shape

(2160, 12)

Answer ; 2160

Q3. Laptop brands

How many laptop brands are presented in the dataset?

In [49]:
a = data.Brand.unique()

In [51]:
a.size

27

In [ ]:
Answer : 27 

Q4. Missing values

How many columns in the dataset have missing values?

In [65]:
data.isnull().sum()

Laptop             0
Status             0
Brand              0
Model              0
CPU                0
RAM                0
Storage            0
Storage type      42
GPU             1371
Screen             4
Touch              0
Final Price        0
dtype: int64

Answer : 3

Q5. Maximum final price

What's the maximum final price of Dell notebooks in the dataset?

In [93]:
dell = data[data['Brand'] == 'Dell']

In [97]:
dell['Final Price'].max()

3936.0

Answer : 3936

Q6. Median value of Screen

Find the median value of Screen column in the dataset.
Next, calculate the most frequent value of the same Screen column.
Use fillna method to fill the missing values in Screen column with the most frequent value from the previous step.
Now, calculate the median value of Screen once again.

1.Find the median value of Screen column in the dataset.

In [109]:
median_screen = data['Screen'].median()
median_screen

15.6

In [113]:
most_frequent = data['Screen'].mode()[0]
most_frequent

15.6

In [105]:
data['Screen'].fillna(most_frequent, inplace=True)

In [111]:
new_median = data['Screen'].median()
new_median

15.6

Answer : No

Q7. Sum of weights
Select all the "Innjoo" laptops from the dataset.
Select only columns RAM, Storage, Screen.
Get the underlying NumPy array. Let's call it X.
Compute matrix-matrix multiplication between the transpose of X and X. To get the transpose, use X.T. Let's call the result XTX.
Compute the inverse of XTX.
Create an array y with values [1100, 1300, 800, 900, 1000, 1100].
Multiply the inverse of XTX with the transpose of X, and then multiply the result by y. Call the result w.
What's the sum of all the elements of the result?

In [115]:
innjoo_laptops = data[data['Brand'] == 'Innjoo']

In [121]:
x = innjoo_laptops[['RAM','Storage','Screen']].to_numpy()

In [123]:
xTx = np.dot(x.T,x)

In [125]:
XTX_inv = np.linalg.inv(xTx)

In [127]:
y = np.array([1100, 1300, 800, 900, 1000, 1100])

In [131]:
w = np.dot(np.dot(XTX_inv,x.T),y)

In [133]:
sum_w = np.sum(w)

In [135]:
print(f"the sum of all the elements :{sum_w}")

the sum of all the elements :91.29988062995588


Answer : 91.30